lib/messenger/slack.rb

```diff
@@ -0,0 +1,68 @@
+require 'httparty'
+require 'json'
+
+class Messenger::Slack
+
+  def self.valid_url?(url)
+    !!matcher(url)
+  rescue NoMethodError
```

Comment:
when are you getting `NoMethodError`? Could `url` be cast to a string in the `matcher` method to avoid needing to deal with that?